src/py/flwr/driver/driver.py

```diff
@@ -35,6 +44,146 @@
+            * server certificate.
+            * server private key.
+
+    Returns
+    -------
+    hist : flwr.server.history.History
+        Object containing training and evaluation metrics.
+
+    Examples
+    --------
+    Starting an insecure server:
+
+    >>> start_driver()
+
+    Starting an SSL-enabled server:
+
+    >>> start_driver(
+    >>>     certificates=Path("/crts/root.pem").read_bytes()
+    >>> )
+    """
+    event(EventType.START_SERVER_ENTER)
+
+    # Parse IP address
+    parsed_address = parse_address(server_address)
+    if not parsed_address:
+        sys.exit(f"Server IP address ({server_address}) cannot be parsed.")
+    host, port, is_v6 = parsed_address
+    address = f"[{host}]:{port}" if is_v6 else f"{host}:{port}"
+
+    # Create the Driver and DriverClientManager if None is provided
+    if client_manager is None:
+        driver = Driver(driver_service_address=address, certificates=certificates)
+        driver.connect()
+
+        client_manager = DriverClientManager(driver=driver, anonymous=False)
+
+    # Initialize the Driver API server and config
+    initialized_server, initialized_config = _init_defaults(
+        server=server,
+        config=config,
+        strategy=strategy,
+        client_manager=client_manager,
+    )
+    log(
+        INFO,
+        "Starting Flower server, config: %s",
+        initialized_config,
+    )
+
+    # Start training
+    hist = _fl(
+        server=initialized_server,
+        config=initialized_config,
+    )
+
+    # Stop the Driver API server
+    client_manager.driver.disconnect()
+
+    event(EventType.START_SERVER_LEAVE)
+
+    return hist
+
+
+def _init_defaults(
+    server: Optional[Server],
+    config: Optional[ServerConfig],
+    strategy: Optional[Strategy],
+    client_manager: ClientManager,
+) -> Tuple[Server, ServerConfig]:
+    # Create server instance if none was given
+    if server is None:
+        if strategy is None:
+            strategy = FedAvg()
+        server = Server(client_manager=client_manager, strategy=strategy)
+    elif strategy is not None:
+        log(WARN, "Both server and strategy were provided, ignoring strategy")
+
+    # Set default config values
+    if config is None:
+        config = ServerConfig()
+
+    return server, config
+
+
+def _fl(
+    server: Server,
+    config: ServerConfig,
+) -> History:
+    # Fit model
+    hist = server.fit(num_rounds=config.num_rounds, timeout=config.round_timeout)
+    log(INFO, "app_fit: losses_distributed %s", str(hist.losses_distributed))
+    log(INFO, "app_fit: metrics_distributed_fit %s", str(hist.metrics_distributed_fit))
+    log(INFO, "app_fit: metrics_distributed %s", str(hist.metrics_distributed))
+    log(INFO, "app_fit: losses_centralized %s", str(hist.losses_centralized))
+    log(INFO, "app_fit: metrics_centralized %s", str(hist.metrics_centralized))
+
+    # Graceful shutdown
+    server.disconnect_all_clients(timeout=config.round_timeout)
+
+    return hist
```

Comment:
Can we import these two functions from `flwr.server.app` to reduce duplication?

Reply:
@danieljanes Should I remove the underscore from the name then?

Reply:
@charlesbvll yes - let's do this in a separate small PR